pandas/io/parsers.py

```diff
@@ -1176,13 +1176,18 @@ def _should_parse_dates(self, i):
         if isinstance(self.parse_dates, bool):
             return self.parse_dates
         else:
-            name = self.index_names[i]
+            if self.index_names is not None:
+                name = self.index_names[i]
```

Comment:
@jorisvandenbossche re-enabled the ``parse_dates`` kw in ``read_excel``, but was failing on this. Note that this by-definition is *only* python engine. (as that is what excel uses).